core/shell.py

```diff
@@ -677,6 +678,20 @@ def Main(lang, arg_r, environ, login_shell, loader, readline):
     func_init.SetGlobalFunc(mem, 'block_as_str', block_as_str)
     func_init.SetGlobalFunc(mem, '_hay', hay_func)
     func_init.SetGlobalFunc(mem, 'len', func_misc.Len())
+    func_init.SetGlobalFunc(mem, '_match', func_eggex.Match(mem))
+    func_init.SetGlobalFunc(mem, '_start', func_eggex.Start(mem))
+    func_init.SetGlobalFunc(mem, '_end', func_eggex.End(mem))
+    func_init.SetGlobalFunc(mem, 'join', func_misc.Join())
+    func_init.SetGlobalFunc(mem, 'maybe', func_misc.Maybe())
+
+    # Do we want to make these methods?
```

Comment:
BTW I think we should make append and pop methods in a follow-on change

And we will should the append builtin to wrap the append method

It's a bit confusing/inconsistent now

Reply:
Ah sure. I can take a pass at this